### PR TITLE
test: add basic tests

### PR DESCRIPTION
**Pull request by colinaaa**
We are replacing `@rstest/core` with `vitest` since Rstest cannot deal with `.ts` file in `node_modules` yet.

## Summary by CodeRabbit

* **New Features**
  * Introduced comprehensive test suites covering various scenarios, including untyped packages, false ESM detection, type resolution failures, fallback conditions, and plugin configuration options.
  * Added new example packages and modules for testing, each with simple exported functions.

* **Bug Fixes**
  * Improved messaging for untyped package detection by prefixing warnings with a standardized tag.

* **Chores**
  * Migrated testing framework to Vitest with coverage reporting and UI support.
  * Updated configuration files for improved dependency and build management.
  * Enhanced `.gitignore` to exclude coverage directories.

* **Tests**
  * Added extensive Vitest-based test coverage for plugin behaviors and edge cases.
  * Removed obsolete or empty test files.

* **Style**
  * Minor formatting improvements for code clarity.

* **Documentation**
  * Added and updated package metadata for new test packages.

### CODE REVIEW

File: test/untyped/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+import { pluginAreTheTypesWrong } from "../../src";
+
+test("should throw when does not contain types", async () => {
+  const rsbuild = await createRsbuild({
+    cwd: import.meta.dirname,
+    rsbuildConfig: {
+      plugins: [pluginAreTheTypesWrong()],
+    },
+  });
```
[the remark this answers is not in the document]
Comment (colinaaa) [reply test/untyped/index.test.ts:16]: @coderabbitai We are setting a global test-timeout in `test.yml`.

File: test/untyped-resolution/index.test.ts [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+  const success = vi.spyOn(logger, "success");
+
+  const { close } = await rsbuild.build();
+
+  expect(success).not.toBeCalled();
+
+  expect(existsSync(path.join(import.meta.dirname, "test-untyped-resolution-0.0.0.tgz"))).toBeFalsy();
+
+  await close();
+});
```
[the remark this answers is not in the document]
Comment (colinaaa) [reply test/untyped-resolution/index.test.ts:154]: @coderabbitai We are setting a global test-timeout in `test.yml`.

File: test/cjs-resolves-to-esm/index.test.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
+import { existsSync } from "node:fs";
+import path from "node:path";
+import { stripVTControlCharacters } from "node:util";
+
+import { createRsbuild, logger } from "@rsbuild/core";
+import { expect, test, vi } from "vitest";
+
```
[the remark this answers is not in the document]
Comment (colinaaa) [reply test/cjs-resolves-to-esm/index.test.ts:7]: @coderabbitai We don't need to restore mocks since we always create a new spy on each test.

File: test/false-cjs/index.test.ts [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+  expect(
+    success.mock.calls.flatMap(call =>
+      call
+        .filter(message => typeof message === "string" && message.includes("[arethetypeswrong]"))
+        .map(stripVTControlCharacters)
+    ),
+  ).toMatchSnapshot();
+
+  expect(existsSync(path.join(import.meta.dirname, "test-false-cjs-0.0.0.tgz"))).toBeFalsy();
+});
```
Comment (colinaaa) [inline test/false-cjs/index.test.ts:95]:
```suggestion
  await close();
});
```
